File: app/src/main/java/ru/vtb/ppcd/example/processor/ErrorHandler.java
package ru.vtb.ppcd.example.processor;

import org.apache.camel.Exchange;
import org.apache.camel.Processor;

public interface ErrorHandler extends Processor {

    default void process(Exchange exchange) {
        Exception caused = exchange.getProperty(Exchange.EXCEPTION_CAUGHT, Exception.class);
        processMessage(exchange, caused);
    }

    void processMessage(Exchange exchange, Exception exception);
}

File: ms-ppcd-example-app/src/main/kotlin/ru/vtb/ppcd/example/processor/ErrorHandler.kt
package org.example.ru.vtb.ppcd.example.processor

import org.apache.camel.Exchange
import org.apache.camel.Processor

interface ErrorHandler : Processor {
    override fun process(exchange: Exchange) {
        exchange.getProperty(
            Exchange.EXCEPTION_CAUGHT,
            Exception::class.java
        ).also { caused -> processMessage(exchange, caused) }
    }

    fun processMessage(exchange: Exchange, exception: Exception)
}